tests/net/mgmt/src/mgmt.c

```diff
@@ -34,6 +35,8 @@ static struct net_mgmt_event_callback rx_cb;
 static struct in6_addr addr6 = { { { 0xfe, 0x80, 0, 0, 0, 0, 0, 0,
 				     0, 0, 0, 0, 0, 0, 0, 0x1 } } };
 
+static char info_data[CONFIG_NET_MGMT_EVENT_INFO_SIZE] = "mgmt event info";
```

Comment:
I just feel this is very error prone with global value. How could user figure out the size requirement for the info? I mean if we want to transfer ipv6 address(s) via info buffer, and some user specific data which smaller information size, then user would need to know that the info size would need to be the largest of these two. I am pretty sure user is not able to figure this out.

I know you abandoned the idea of having the info buffer / listener, but would it be possible to do it like that?

Reply:
The event number will dictate the data:
- if up/down: none
- ipv4 add/del: struct in_addr
- ipv6 add/del: struct in6_addr
- etc...

What user specific data?

Reply:
Because net_mgmt API is open, you can have your own events created and that event might have info which has some size requirements.

Reply:
Well, up to the Kconfig option to fit the required size then. New event means new structure with a difference size, fine as long as the Kconfig option is properly set. (I could put a NET_ERR or assert actually in net_mgmt core that the notification info cannot fit into the data buffer).

net_mgmt is not netlink, it should not be used as an ipc or aggregated nested data or else. We just don't have the luxury to do that, and it's anyway not the idea behind net_mgmt. It's really meant for net core, bare minimum notification and that's it. I guess we should enforce that in the documentation.

The info buffer per listener is not possible with net_mgmt design: when the event is "notified" it's queued directly and has no knowledge of any listener. It's only when the notifier thread (internal to net_mgmt) runs that it goes through queued event and relate them to listeners relevantly. I made it like that for performance and logical purposes. (a listener might be added, right before the notifier thread gets to run for instance, etc..).
Thus why the info data is passed once from the notifier, and passed as const to the listener.